### PR TITLE
groceries app design

### PR DESCRIPTION
**Pull request by qm-jagadish**

### CODE REVIEW

File: GroceriesApp/GroceryappDesign.sql [hunk 1]
```diff
@@ -0,0 +1,208 @@
+--drop database GroceriesAppdb
+
+CREATE DATABASE GroceriesAppdb;
+
+--creating table
+
+CREATE TABLE Register (
```
Comment (qm-afrin) [inline GroceriesApp/GroceryappDesign.sql]: Change the table name to UserDetails

File: GroceriesApp/GroceryappDesign.sql [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+INSERT INTO Register (
+    FirstName, LastName, Email, [Password], ConfirmPassword, PhoneNumber,[Address], City, [State], PinCode
+) VALUES
+    ('John', 'Doe', 'john.doe@example.com', 'password123', 'password123', '+1234567890', '123 Main St', 'New York', 'NY', '10001'),
+    ('Jane', 'Smith', 'jane.smith@example.com', 'password123', 'password123', '+0987654321', '456 Maple Ave', 'Los Angeles', 'CA', '90001');
+
+
+	
+CREATE TABLE [Login](
```
Comment (qm-afrin) [inline GroceriesApp/GroceryappDesign.sql]: Remove login table and add the required columns in UserDetails table.

File: GroceriesApp/GroceryappDesign.sql [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+CREATE TABLE Products(
+ ProductsId INT IDENTITY(1,1),
+ ProductSupplierId INT,
+ ProductName VARCHAR(20) NOT NULL,
+ ProductPrice MONEY NOT NULL,
+ ProductDescription VARCHAR(50) NOT NULL,
+ ProductQuantity INT NOT NULL,
+ CategoryId INT,
+ CONSTRAINT FK_Products_Catergories FOREIGN KEY(CategoryId) REFERENCES Categories(CategoryId),
+ CONSTRAINT FK_Products_Products_Supplier FOREIGN KEY(ProductSupplierId) REFERENCES Products_Supplier(ProductSupplierId),
```
Comment (qm-afrin) [inline GroceriesApp/GroceryappDesign.sql]: First declare the Products_Supplier table and then use it.

File: GroceriesApp/GroceryappDesign.sql [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+VALUES
+(1, 'Apple', 0.50, 'Fresh Red Apple', 100, 1),
+(2, 'Carrot', 0.20, 'Organic Carrot', 200, 2),
+(5, 'Cheese', 2.00, 'Cheddar Cheese', 30, 3);
+
+SELECT * FROM Products;
+
+
+ -- Create the Supplier table
+CREATE TABLE SupplierDetails (
```
Comment (qm-afrin) [inline GroceriesApp/GroceryappDesign.sql]: Remove SupplierDetails table and declare where the user belongs to which type of user in UserDetails table.

**Review by qm-afrin — CHANGES_REQUESTED** [review]

File: GroceriesApp/GroceryappDesign.sql [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+    LastName VARCHAR(20) NOT NULL,
+    Email VARCHAR(40) , -- Unique email
+    [Password] VARCHAR(20) NOT NULL,
+    ConfirmPassword VARCHAR(20) NOT NULL,
+    PhoneNumber VARCHAR(15) NOT NULL, -- Unique phone number
+    [Address] VARCHAR(60) NOT NULL, 
+    City VARCHAR(30) NOT NULL,
+    [State] VARCHAR(30) NOT NULL,
+    PinCode VARCHAR(10) NOT NULL,
+	UserType VARCHAR(40) NOT NULL,
```
Comment (qm-afrin) [inline GroceriesApp/GroceryappDesign.sql]: What is the use of this column? Remove it.

File: GroceriesApp/GroceryappDesign.sql [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+(1,10.00, 'Processing'),   -- Order by John Doe for Apple
+(2,20.00, 'Shipped'),      -- Order by Jane Smith for Carrot
+(3,30.00, 'Delivered');   -- Order by John Doe for Cheese
+
+
+-- Verify the data insertion
+SELECT * FROM Orders;
+
+
+    CREATE TABLE Payment (
```
Comment (qm-afrin) [inline GroceriesApp/GroceryappDesign.sql]: How are you differentiating this payment is related to which customer?

File: GroceriesApp/GroceryappDesign.sql [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+(1, 'Apple', 0.50, 'Fresh Red Apple', 100, 1),
+(2, 'Carrot', 0.20, 'Organic Carrot', 200, 2),
+(5, 'Cheese', 2.00, 'Cheddar Cheese', 30, 3);
+
+-- Verify the data insertion
+SELECT * FROM Products;
+
+
+
+CREATE TABLE [Orders](
```
Comment (qm-afrin) [inline GroceriesApp/GroceryappDesign.sql:87]: How are you differentiating this order belongs to which customer and what they bought?

File: GroceriesApp/GroceryappDesign.sql [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+    ( 'Fruits'),
+    ( 'Vegetables'),
+    ( 'Dairy');
+
+	
+-- Verify the data insertion
+SELECT * FROM Categories;
+
+CREATE TABLE Products(
+ ProductsId INT IDENTITY(1,1),
```
Comment (qm-afrin) [inline GroceriesApp/GroceryappDesign.sql]: What is the difference between Products table and Products_Supplier table?

Both have the same information inserted.
For Example: Apple fruit is inserted as the first record in both the tables. Then what is the use of creating two tables?

File: GroceriesApp/GroceryappDesign.sql [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+--creating table
+
+DROP TABLE UserDetails
+CREATE TABLE  UserDetails(
+    RegisterId INT IDENTITY(1,1),
+    FirstName VARCHAR(20) NOT NULL,
+    LastName VARCHAR(20) NOT NULL,
+    Email VARCHAR(40) , -- Unique email
+    [Password] VARCHAR(20) NOT NULL,
+    ConfirmPassword VARCHAR(20) NOT NULL,
```
Comment (rojamaddipati) [inline GroceriesApp/GroceryappDesign.sql]: you can remove confirmpassword  column
No need of having confirmpassword column we can validate from UI

**Pull request merged**